core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3648,12 +3647,8 @@ class KafkaApis(val requestChannel: RequestChannel,
         deleteShareGroupOffsetsResponseTopics.add(
           new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponseTopic()
             .setTopicName(topic.topicName)
-            .setPartitions(topic.partitions.map(partition => {
-              new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponsePartition()
-                .setPartitionIndex(partition)
-                .setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code)
-                .setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message())
-            }).toList.asJava)
+            .setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code)
+            .setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message())
```

Comment:
It seems to me that the code below that handles the situation in which the set of authorized topics is empty actually sends an entirely empty response. The `deleteShareGroupOffsetsResponseTopics` ought to be in there somewhere :)